tools/publishIsolate.ts

```diff
@@ -0,0 +1,76 @@
+import fs from 'fs';
+import path from 'path';
+import childProcess from 'child_process';
+import getNpmPackageVersion from 'get-npm-package-version';
+
+async function main() {
+  const args = process.argv;
+  const npmAuthToken = args[0];
+  const isolatePackageRoot = './dist/isolate';
+
+  const publishPackageList: string[] = [];
+
+  // get directory listing
+  const files = fs.readdirSync(isolatePackageRoot);
+  for (const file of files) {
+    const filePath = path.join(isolatePackageRoot, file);
+    const stat = fs.statSync(filePath);
+    if (stat.isDirectory()) {
+      // get package.json
+      const packageJsonPath = path.join(filePath, 'package.json');
+      const packageJson = JSON.parse(fs.readFileSync(packageJsonPath, 'utf8'));
+      // get version
+      const version = getNpmPackageVersion(packageJson.name);
+
+      if (version === null) {
+        throw new Error(`Package ${packageJson.name} not found on npm`);
+      }
+
+      // compare version
+      if (compareVersions(version, packageJson.version) < 0) {
+        publishPackageList.push(packageJson.name);
+      }
+    }
+  }
+
+  // publish packages
+  for (const packageName of publishPackageList) {
```

Comment:
Use Promise.all() please.

Reply:
sorry I don't understand.

Reply:
Await will block the thread.

You can use `Promise.all` or `Promise.allSettled` to make it be a real async function instead.

```typescript
await Promise.all(publishPackageList.map(packageName => new Promise((resolve, reject) => { ... })));
```

Reply:
However, it will publish multiple packages at the same time, mixing the output messages. Is that okay?

Reply:
Why?

`console.log` will only run in resolved Promise.

Reply:
I think I was mistaken.